### ISSUE TITLE
Automatically convert PascalCase to --kebab-case args

### ISSUE THREAD

**Issue by ryanc414** [alexflint/go-arg#143]
First of all, just want to say I really like this project! After using the Rust structopt package for a while, I was really happy to find a very similar package existed in the Go ecosystem.

I think where possible, go-arg should always try and do the most sensible thing by default without having to always specify everything explicitly in the tags. With that in mind, I notice that currently the default names of CLI flags seem to be generated simply by converting the struct field to lowercase - e.g. a DryRun field becomes --dryrun. It would be nice if this could be automatically generated as --dry-run instead, with the extra dash for readability. Since PascalCase is the dominant convention for naming things in Go, this seems to map naturally to the kebab-case often used for CLI args.

Of course, PascalCase is a convention and not a requirement. I think it would be reasonable to revert to the simple lowercase name without dashes for any field name that contains two or more consecutive capital letters (e.g. UserID would become --userid by default and not --user-i-d).

Do you have any thoughts on implementing this, or accepting a PR on it?

**Comment by alexflint** [comment]
Thank you @ryanc414. I'm happy the package is helpful for you and grateful for you taking the time to include these kind words.

I did consider something like this when I first implemented go-arg but decided it would be too unpredictable in cases with single-letter words, such as `PickARandomNumber`, so I decided to go with simplicity over always doing the most sensible thing. But as you say, there are some drawbacks to my choice.

Now that it's been done this way, it's difficult to change the default behavior, since that would break a lot of existing users' code.

Thank you for engaging.
